src/lib.rs

```diff
@@ -138,6 +138,8 @@
 pub mod converter;
 pub mod suffix_array;
 
+// TODO: public module for now so as not to conflict with the other builder
+pub mod builder;
```

Comment:
Now we dropped the builder from #36, so we can expose the builder structs instead?

Reply:
Yes, good point. I will only make the builder public and then make all other construction methods private, as you had also already planned. Once I do that tomorrow I will merge.